mlos_bench/mlos_bench/optimizers/base_optimizer.py

```diff
@@ -68,25 +68,30 @@ def __init__(self,
         self._seed = int(config.get("seed", 42))
         self._in_context = False
 
-        experiment_id = self._global_config.get('experiment_id')
+        experiment_id = self._global_config.get("experiment_id")
```

Comment:
here and everywhere: let's separate cosmetic fixes from functional changes. mixing the two bloats up the diff and makes reviewing PRs much harder. Please keep the essential functionality in this PR and make the diff as small as possible and move all code annotations, style, and docstring improvements to a separate PR. having more PRs is good for your github karma! 😄

Reply:
Sounds good! #744 should do just the style changes

Reply:
Yeah, @motus @jsfreischuetz and I already talked about that.  #744 is the start of that, but I think will also need some additional work.  #746 unearthed some related topics around `pyproject.toml` settings vs. `setup.cfg` for using `black`.

Am currently working on improvements for all of those and will send a split out series of PRs soon for:
1. pyproject.toml settings and build related improvements
  - annoyed at this one since there's no "include" feature, so it means duplicating some content across files it looks like
2. adding `black` and `isort` `Makefile` rules, `vscode` settings, configs, etc.
3. requiring them in the build (including all of the reformatting of all files)
4. adding the revision from 3 in the `.gitrevisions` list to ignore it from most `git blame` types of analysis